network/gossip/libp2p/test/helper.go

```diff
@@ -69,7 +70,14 @@ func createNetworks(log zerolog.Logger, mws []*libp2p.Middleware, ids flow.Ident
 		me := &mock.Local{}
 		me.On("NodeID").Return(ids[i].NodeID)
 		me.On("NotMeFilter").Return(flow.IdentityFilter(filter.Any))
-		net, err := libp2p.NewNetwork(log, json.NewCodec(), identities, me, mws[i], csize, tops[i], metrics)
+		var state protocol.ReadOnlyState
+		if dryrun {
+			state = generateStateSnapshot(ids)
+		} else {
+			identities := make(flow.IdentityList, 0)
+			state = generateStateSnapshot(identities)
+		}
```

Comment:
When `dryrun` is specified we use the passed-in `ids` and otherwise we generate an empty list? Kind of seems backward to me 🤔

Reply:
so we first start the network with empty identities if dryrun is false, then when the network is up and running we modify the identities with the actual identities with real ip and port. But, as you pointed out - this is confusing and also does not depict the real usage where the ids will already have IP and port. 
I have removed `helper.go` all together now.